modules/user_profiles/src/Commands/UserProfilesCommands.php

```diff
@@ -0,0 +1,197 @@
+            return;
+        }
+        
+        foreach ($flagged_items as $flagged_item) {
+            $term = \Drupal\taxonomy\Entity\Term::load($flagged_item);
+            $title = $term->get('name')->value;
+
+            $this->output()->writeln("  from-user has $flag_name '$title'");
+
+            // Check if already flagged. If not, set the flag.
+            $flag_service = \Drupal::service('flag');
+            $flag = $flag_service->getFlagById($flag_name);
+            $flag_status = $flag_service->getFlagging($flag, $term, $user_to);
+            if (!$flag_status) {
+                $this->output()->writeln("    Add $flag_name '$title' to to-user");
+                $flag_service->flag($flag, $term, $user_to);
+            } else {
+                $this->output()->writeln("    To-user already has $flag_name '$title'");
+            }
+        }
+
+    }
+
+    private function mergeAfffinityGroups($user_from, $user_to)
+    {
+
+        $this->output()->writeln("Merging affinity groups");
+
+        // get user_to's blocked ag taxonomy ids
+        $user_blocked_tid_array = $user_to->get('field_blocked_ag_tax')->getValue();
+        $user_blocked_tids = [];
+        foreach ($user_blocked_tid_array as $user_blocked_tid) {
+            $user_blocked_tids[] = $user_blocked_tid['target_id'];
+        }
+        // $this->output()->writeln("  user-to blocked ag tids: " . implode(' ', $user_blocked_tids));
+
+        // get all the affinity groups of $user_from
+        $query = \Drupal::database()->select('flagging', 'fl');
+        $query->condition('fl.uid', $user_from->id());
+        $query->condition('fl.flag_id', 'affinity_group');
+        $query->fields('fl', ['entity_id']);
+        $ag_ids = $query->execute()->fetchCol();
+
+        if ($ag_ids == NULL) {
+            $this->output()->writeln("  from-user is not a member of any affinity groups");
+            return;
+        }
+        // for each affinity group id, add user_to to that affinity group
+        // $this->output()->writeln("  from-user ag ids: " . implode(' ', $ag_ids));
+        foreach ($ag_ids as $ag_id) {
+            $this->addUserToAG($user_to, $ag_id, $user_blocked_tids);
+        }
+    }
+
+    /**
+     * Add a user to an affinity group (unless on the users's block list)
+     */
+    private function addUserToAG(UserInterface $to_user, $ag_id, $user_blocked_tids)
+    {
+        // get the node id of the affinity group
+        $query = \Drupal::database()->select('taxonomy_index', 'ti');
+        $query->condition('ti.tid', $ag_id);
+        $query->fields('ti', ['nid']);
+        $affinity_group_nid = $query->execute()->fetchCol();
+
+        if (!isset($affinity_group_nid[0])) {
+            // not sure how or if this could happen, or what it would mean, but Miles' code in 
+            // CommunityPersonaController.php line 36 also ignores this condition
```

Comment:
This would be the case if there is not a taxonomy term associated with the Affinity Group.